lib/solidus_graphql_api/types/ransack_json.rb

```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+
+module SolidusGraphqlApi
+  module Types
+    class RansackJson < GraphQL::Types::JSON
+      def self.coerce_input(value, _ctx)
+        value.each do |k, v|
+          begin
+            case v
+            when String
+              class_name, item_id = GraphQL::Schema::UniqueWithinType.decode(v)
+              ::Object.const_get(class_name)
+              value[k] = Integer(item_id)
+            when Array
+              v.each_with_index do |child_field, index|
+                class_name, item_id = GraphQL::Schema::UniqueWithinType.decode(child_field.to_s)
+                ::Object.const_get(class_name)
+                v[index] = Integer(item_id)
+              end
+            end
+          rescue ArgumentError, NameError
```

Comment:
Can we move this `rescue`s around the `decode` line? Something like this:

```ruby
begin
  class_name, item_id = GraphQL::Schema::UniqueWithinType.decode(v)
rescue ArgumentError, NameError
  next
end
```

In this way we avoid rescuing exceptions that we don't want to rescue